packages/mobile/src/components/now-playing-drawer/PlayButton.tsx

```diff
@@ -39,15 +44,38 @@ const useAnimations = makeAnimations(({ palette, type }) => {
     'layers.2.shapes.0.it.1.c.k': palette.primary
   })
 
-  return [ColorizedPlayIcon, ColorizedPauseIcon]
+  const ColorizedSpinnerIcon = IconLoadingSpinner
+  return [ColorizedPlayIcon, ColorizedPauseIcon, ColorizedSpinnerIcon]
 })
 
 type PlayButtonProps = Omit<AnimatedButtonProps, 'iconJSON' | 'iconIndex'>
 
 export const PlayButton = ({ isActive, ...props }: PlayButtonProps) => {
   const isPlaying = useSelector(getPlaying)
+  const [showBufferingState, setShowBufferingState] = useState(false)
+
+  const isBuffering = useSelector(getBuffering)
+
+  // To prevent "flashes" of buffering states & not show it at all when tracks are playing quickly,
+  // we only show the buffering spinner state if a minimum amount of time has passed
+  useEffect(() => {
```

Comment:
This looks good. Iirc we might have a ready made timeout hook/ react use might have something.